### PR TITLE
[FUSEQE-4065] Add SQS tests

### PR DESCRIPTION
**Pull request by avano**
//test: `@sqs`

### CODE REVIEW

**Comment by asmigala** [comment]
We discussed adding a binary jar file offline, but one thing I'd still like to see is info in the readme on where does this jar come from, what it contains and how to rebuild it if necessary.

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName;
+import software.amazon.awssdk.services.sqs.model.QueueDoesNotExistException;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequestEntry;
+
+@Slf4j
+@Component
+@Lazy
+public class SQSUtils {
+    private static final String AWS_ACCOUNT_ID = "928544467084";
```
Comment (asmigala) [inline utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java]: Shouldn't this also be in credentials.json?

File: utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    /**
+     * Sends one or more messages as batch message. The messages have generated IDs and the body as specified.
+     *
+     * @param messageContents bodies of the messages to send
+     */
+    public void sendMessages(String queueName, List<String> messageContents) {
+        log.debug("Sending batch message to queue " + queueName + " with messages: " + messageContents.toString());
+        SendMessageBatchRequest.Builder sendMessageBatchRequestBuilder = SendMessageBatchRequest.builder().queueUrl(getQueueUrl(queueName));
+        List<SendMessageBatchRequestEntry> entries = new ArrayList<>();
+        for (int i = 1; i < messageContents.size() + 1; i++) {
```
Comment (asmigala) [inline utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java]: I'd probably prefer a standard loop starting at 0 and changing the mod check to `i % 10 == 9` or even `(i + 1) % 10 == 0`. I think there's less of a chance of introducing bugs when somebody needs to touch this code in the future.

(also easier at a glance to see what's happening)

Maybe even use something like `Iterables.partition` to split it to chunks beforehand and avoid this loop altogether

File: utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java (hunk 1 above)
Comment (avano) [reply utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java]: yup, I already sent a PR to ci-ops-fuse

File: rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java [hunk 3]
```diff
@@ -164,6 +168,32 @@ public void createKafkaStep(String action, String topic) {
         super.createStep();
     }
 
+    @When("^create SF \"([^\"]*)\" action step with properties$")
+    public void createSfStepWithActionAndProperties(String action, DataTable props) {
+        super.addProperty(StepProperty.CONNECTOR_ID, RestTestsUtils.Connector.SALESFORCE.getId());
```
[the remark this answers is not in the document]
Comment (avano) [reply rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java]: there is a common method in super class that creates the step from given properties https://github.com/syndesisio/syndesis-qe/blob/master/rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/AbstractStep.java#L71

File: utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java (hunk 2 above)
Comment (avano) [reply utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java]: cool, I didn't know the `Iterables.partition`, I'll edit

File: rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (asmigala) [reply rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java]: Sure, but you don't need the `super`, right? You can access it just by `addProperty`, no?

File: utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java (hunk 1 above)
Comment (asmigala) [reply utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java]: So you'll remove it from here?

Comment (avano) [reply utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java]: yes yes yes

File: rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (avano) [reply rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java]: yes, it is not needed, but I find it more readable, especially when going through the code somewhere else than in the IDE

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
@asmigala only the ui tests failed

File: rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (asmigala) [reply rest-tests/src/test/java/io/syndesis/qe/rest/tests/steps/flow/ConnectionSteps.java]: iguess

**Pull request merged**